Source/Diagnostics/Diagnostics.H

```diff
@@ -18,99 +18,84 @@ class Diagnostics
 {
 public:
     Diagnostics (int i, std::string name);
-    ~Diagnostics ();
+    /** Virtual Destructor to handle clean destruction of derived classes */
+    virtual ~Diagnostics () ;
     /** Pack (stack) all fields in the cell-centered output MultiFab m_mf_output.
-     * Fields are computed (e.g., cell-centered) on-the-fly using a functor.*/
+     * Fields are computed (e.g., cell-centered or back-transformed)
+       on-the-fly using a functor.*/
     void ComputeAndPack ();
     /** Flush m_mf_output and particles to file. */
-    void Flush ();
-    /** Flush raw data */
-    void FlushRaw ();
+    virtual void Flush (){};
+    /** Flush raw data (applies only to full diagnostics)*/
+    virtual void FlushRaw (){};
     /** Initialize pointers to main fields and allocate output multifab m_mf_output. */
-    void InitData ();
+    virtual void InitData (){};
```

Comment:
see below `InitializeFieldFunctors` 
```suggestion
    virtual void InitData () = 0;
```